engine.go

```diff
@@ -388,15 +412,27 @@ func (e *RTCEngine) waitUntilConnected() error {
 }
 
 func (e *RTCEngine) ensurePublisherConnected(ensureDataReady bool) error {
+	e.pclock.Lock()
```

Comment:
is it cleaner/simpler to grab the transport under lock? vs having to unlock during waitUntilConnected() ?